src/libcmd/repl.cc

```diff
@@ -111,23 +111,23 @@ NixRepl::~NixRepl()
     write_history(historyFile.c_str());
 }
 
-std::string runNix(Path program, const Strings & args,
+void runNix(Path program, const Strings & args,
     const std::optional<std::string> & input = {})
 {
     auto subprocessEnv = getEnv();
     subprocessEnv["NIX_CONFIG"] = globalConfig.toKeyValue();
 
-    auto res = runProgram(RunOptions {
+    int res = runProgramWithoutCapturingStdout(RunOptions {
```

Comment:
Apart from the very untelling name, I think you could directly use `runProgram2` here, right?